cluster/ci/config/prow/jobs/master-sidecar.yaml

```diff
@@ -0,0 +1,103 @@
+kind: Template
+apiVersion: template.openshift.io/v1
+
+parameters:
+- name: JOB_NAME_SAFE
+  required: true
+- name: IMAGE_CONTROL_PLANE
+  required: true
+- name: LOCAL_IMAGE_TEST_BIN
+  required: true
+- name: COMMAND
+  required: true
+- name: NAMESPACE
+  required: true
+
+objects:
+
+# The e2e pod spins up a cluster, runs e2e tests, and then cleans up the cluster.
+- kind: Pod
+  apiVersion: v1
+  metadata:
+    name: ${JOB_NAME_SAFE}
+    namespace: ${NAMESPACE}
+  spec:
+    restartPolicy: Never
+    activeDeadlineSeconds: 7200
+    terminationGracePeriodSeconds: 600
+    volumes:
+    - name: artifacts
+      emptyDir: {}
+    - name: shared-tmp
+      emptyDir: {}
+
+    containers:
+
+    # Once admin.kubeconfig exists, executes shared tests
+    - name: test
+      image: ${LOCAL_IMAGE_TEST_BIN}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        trap 'touch /tmp/shared/exit' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # wait until the master job creates admin.kubeconfig
+        while true; do
+          if [[ ! -f /tmp/shared/admin.kubeconfig ]]; then
+            sleep 5 & wait
+            continue
+          fi
+          break
+        done
+        echo "Found kubeconfig"
+        export KUBECONFIG=/tmp/shared/admin.kubeconfig
+
+        ${COMMAND}
+
+    # Start a standalone master
+    - name: control-plane
+      image: ${IMAGE_CONTROL_PLANE}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      env:
+      - name: TYPE
+        value: ${CLUSTER_TYPE}
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+        function teardown() {
+          set +e
+          echo "Gathering artifacts ..."
+        }
+        trap 'teardown' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        ( 
+          while true; do
+            if [[ -f /tmp/exit ]]; then
+              echo "Shutting down"
+              kill $$
+            fi
+            sleep 5 & wait
+          done
+        ) &
+
+        mkdir /tmp/cluster && cd /tmp/cluster
```

Comment:
You were thinking about something

Reply:
Wanted a working directory for crap that wouldn't show up in artifacts or shared content.